onnxruntime/hosting/http/session.cc

```diff
@@ -96,23 +94,42 @@ template <typename Body, typename Allocator>
 void HttpSession::HandleRequest(http::request<Body, http::basic_fields<Allocator> >&& req) {
   HttpContext context{};
   context.request = std::move(req);
-
   // TODO: set request id
-  std::string path = context.request.target().to_string();
-  std::string model_name, model_version, action;
-  handler_fn func;
-  http::status status = routes_->ParseUrl(context.request.method(), path, model_name, model_version, action, func);
 
-  if (http::status::ok == status && func != nullptr) {
-    func(model_name, model_version, action, context);
-  } else {
-    context.response.result(status);
+  auto status = ExecuteUserFunction(context);
+
+  if (status != http::status::ok) {
+    routes_->on_error(context);
   }
 
   context.response.keep_alive(context.request.keep_alive());
   context.response.prepare_payload();
   return Send(std::move(context.response));
 }
 
+http::status HttpSession::ExecuteUserFunction(HttpContext& context) {
```

Comment:
I think you may want to mention in the header that the context will be updated in the context with status and error message in case something wrong.

Reply:
Done.